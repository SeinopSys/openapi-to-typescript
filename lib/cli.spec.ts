import execa from 'execa'
import * as yargs from 'yargs';

describe('openapi-to-typescript CLI', () => {
  it('should --help', async () => {
    const executed = await runCLI('--help')
    expect(executed.stdout).toMatchSnapshot()
  })

  describe('--input -i', () => {
    it('should output typescipt code to stdout', async () => {
      const executed = await runCLI('-i ./fixtures/petstore.json')
      expect(executed.stdout).toContain('API_URL = ')
      expect(executed.stdout).toContain('ShowPetById')
    })
  })

  describe('--server -s', () => {
    it('should set the API url to the one provided', async () => {
      const executed = await runCLI('-i ./fixtures/petstore.json -s some://custom.api')
      expect(executed.stdout).toContain('API_URL = "some://custom.api"')
    })
  })
})

function runCLI(arg: string, opts?: any) {
  return execa('ts-node', ['./lib/cli', ...arg.split(/\s+/g)], {
    ...opts,
    env: { LC_ALL: 'en_US' },
  })
}
