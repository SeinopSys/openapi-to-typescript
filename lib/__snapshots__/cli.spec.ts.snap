// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openapi-to-typescript CLI should --help 1`] = `
"cli

Write generated output to a file

Commands:
  cli generate  Write generated output to a file                       [default]

Options:
      --help     Show help                                             [boolean]
      --version  Show version number                                   [boolean]
  -i, --input    File to read from                           [string] [required]
  -o, --output   Output name to write to                                [string]
  -s, --server   Server URL to use (defaults to the first one in the OpenAPI
                 spec)                                                  [string]
  -v, --verbose                                       [boolean] [default: false]"
`;
